%!TEX root = ../main.tex
%******************************
%	 Chapter 2
%*****************************

\chapter{Addressing the mean-variability dependency in scRNA-Seq data}  

\graphicspath{{"../../Dropbox (Cambridge  University)/Figures_for_thesis/Chapter2/"}}

\vfill

\begin{Abstract}
\hspace{-5mm} As shown above, cell-to-cell transcriptional variability in otherwise homogeneous cell populations plays an important role in immune activation and increases with age. Single-cell RNA sequencing can characterise this variability in a transcriptome-wide manner. However, the confounding between variability and mean expression estimates hinders meaningful comparison of expression variability between cell populations. To address this problem, we introduce a statistical approach that extends the BASiCS framework to derive a residual measure of variability that is not confounded by mean expression. This measure is used to test changes in variability in parallel to changes in mean expression on a gene-specific level. With this method, we assessed changes in variability for genes responding to CD4\plus{} T cell activation and detect a synchronisation of biosynthetic machinery components. Furthermore, cytokines such as Il2 that support the activation of surrounding cells by paracrine signalling are heterogeneously up-regulated upon immune activation. When profiling more subtle transcriptional changes during CD4\plus{} T cell differentiation, we detect opposing patterns of changes in variability between \textit{Tbx21} and \textit{Cxcr5}, which are markers for Th1 and Tfh cells, indicating a delayed commitment process throughout differentiation. Finally, we confirmed the applicability of the newly extended BASiCS model to droplet-based scRNA-Seq data which is necessary for the subsequent chapter.
\end{Abstract}

\vfill

\newpage

\begin{Comment}
\hspace{-3mm} \textbf{Declaration} I worked on this project in close collaboration with Catalina Vallejos, Arianne Richard, Sylvia Richardson and John Marioni. In this project, I extended an existing Bayesian framework (BASiCS) that was developed to model scRNA-Seq data. Besides calculating the underlying math and programming the model, I helped re-write the BASiCS R package, which is now published on \href{https://bioconductor.org/packages/release/bioc/html/BASiCS.html}{Bioconductor}. Catalina Vallejos co-supervised me and introduced me to principles of Bayesian statistics. Arianne Richard helped with the interpretation of biological data and helped writing the manuscript. Sylvia Richardson provided statistical help on a part of the project that is not presented here. John Marioni co-supervised me and suggested analysis to be performed. Catalina Vallejos, John Marioni and I designed the study. Catalina Vallejos, John Marioni and I wrote the manuscript. I kindly thank Dominic Gr\"un for providing the smFISH data matched to mESC scRNA-Seq data. The study has been published as:\\

Nils Eling, Arianne C. Richard, Sylvia Richardson, John C. Marioni, Catalina A. Vallejos. Robust expression variability testing reveals heterogeneous T cell responses. \emph{Cell Systems}, In press, 2018 
\end{Comment}

\begin{figure}[hb]
\centering    
\includegraphics[width=0.5\textwidth]{GraphicalAbstract.png}
\end{figure}

\newpage

% Include different main sections of the first chapter
\input{Chapter2/Chapt2_files/intro}
\newpage
\input{Chapter2/Chapt2_files/results}
\newpage
\input{Chapter2/Chapt2_files/discussion}

