%!TEX root = ../chapter2.tex
%******************************
%	 Introduction 
%*****************************

\section{Introduction}

Heterogeneity in gene expression within a population of single cells can arise from a variety of factors (see \textbf{Box 1} on page \pageref{box1}). In a seemingly homogeneous population of cells,  unstructured expression heterogeneity can be linked to intrinsic or extrinsic noise \citep{Elowitz2002}. Changes in physiological cell states (e.g.~cell cycle) represent extrinsic noise, which has been found to influence expression variability within cell populations \citep{Keren2015, Buettner2015, Kiviet2014}. Intrinsic noise can be linked to epigenetic diversity \citep{Smallwood2014}, chromatin accessibility \citep{Buenrostro2015}, and the genomic content of single genes \citep{Hornung2012}.  \\ 

Single-cell RNA sequencing generates transcriptional profiles of individual cells which allows the study cell-to-cell heterogeneity on a transcriptome-wide \citep{Grun2014} and single gene level \citep{Goolam2016}. Consequently, this technique can be used to profile unstructured cell-to-cell variation in gene expression within and between homogeneous cell populations (i.e.~where no distinct cell sub-types are present). As shown in \textbf{Section \ref{sec1:activation}}, transcriptional noise decreases during immune activation. Ageing on the other hand destabilises the immune response, which manifests itself in the form of increased transcriptional noise. Furthermore, increasing evidence suggests that this heterogeneity plays an important role in development \citep{Chang2008}. For instance, molecular noise was shown to increase before cells commit to lineages during differentiation \citep{Mojtahedi2016}, while the opposite is observed once an irreversible cell state is reached \citep{Richard2016}. A similar pattern occurs during gastrulation, where expression noise is high in the uncommitted inner cell mass at E3.5 compared to the epiblast at E4.5 and where an increase in heterogeneity is observed when cells exit the pluripotent state and form the uncommitted epiblast at E6.5 (see \textbf{Fig.~\ref{fig0:noise_development}} and \citep{Mohammed2017}). \\

Motivated by scRNA-Seq, recent studies have extended traditional differential expression analyses to explore more general patterns that characterise differences between cell populations \citep{Korthauer2016}. As described in \textbf{Section \ref{sec0:BASiCS}} and \textbf{\ref{sec1:BASiCS}}, BASiCS \citep{Vallejos2015BASiCS,Vallejos2016} introduced a probabilistic tool to assess differences in cell-to-cell heterogeneity between two or more cell populations. To meaningfully assess changes in biological variability across the entire transcriptome, one strong confounding effect must be taken into account: differential variability between populations that is driven by changes in mean expression. This arises because biological noise is negatively correlated with protein abundance \citep{Bar-Even2006, Newman2006, Taniguchi2011} or mean RNA expression (see \textbf{Section \ref{sec0:BASiCS}} and \citep{Brennecke2013, Antolovic2017}). To acknowledge the variance-mean relationship, the initial version of BASiCS restricted differential variability testing to those genes with equal mean expression across populations (see \textbf{Section \ref{sec1:BASiCS}}). \\

Previous studies derived measures of transcriptional variability that are independent of mean expression. These approaches ranged from a simple linear regression between the logarithm of the coefficient of variation $\log_2$(CV) and the $\log_2$(mean expression) \citep{Wu2017} to more elaborate models as described by Gr\"un \emph{et al.}, 2014. Their model aims to capture (i) the Poissonian sampling noise for lowly expressed transcripts and (ii) differences in total transcript abundance between cells for highly expressed genes. The mixture of these effects introduces a non-linear relationship between mean expression and the CV. The model also captures technical noise by incorporating reads from technical spike-in RNA. In this case, the number of transcripts available for sequencing is Gamma distributed due to variation in capture efficiency. The sequencing process on the other hand is a Poisson process \cite{Marioni2008}. The combination of these distribution forms a negative binomial which models the expression counts of all biological genes best \cite{Grun2014}. A non-parmateric strategy to model the mean-variance relationship was proposed by Kolodziejczyk \emph{et al.}, 2015, where the mean-independent measure of variability is the distance between the CV$^2$ and a rolling median along mean expression \citep{Kolodziejczyk2015cell}.\\

In this chapter, we extend the statistical model in BASiCS by implementing a more general approach to account for this confounding effect. By incorporating a flexible, non-linear regression trend, we derive a residual measure of cell-to-cell transcriptional variability that is not confounded by mean expression. This is used to define a probabilistic rule to robustly highlight changes in variability, even for differentially expressed genes. Unlike previous approaches that derive point estimates of residual variability, our approach directly performs gene-specific statistical testing between two conditions using a readily available measure of uncertainty. \\

Using our approach, we identify a synchronisation of  biosynthetic machinery components in CD4\plus{} T cells upon early immune activation as well as an increased variability in the expression of genes related to CD4\plus{} T cell immunological function.
Furthermore, we detect evidence of early cell fate commitment of CD4\plus{} T cells during malaria infection characterised by a decrease in \textit{Tbx21} expression heterogeneity and a rapid collapse of global transcriptional variability after infection. These results highlight biological insights into T cell activation and differentiation that are only revealed by jointly studying changes in mean expression and variability.
