%!TEX root = ../main.tex
%******************************
%	 Chapter 3
%*****************************

\chapter{Transcriptional dynamics during spermatogenesis at single-cell resolution}  

\graphicspath{{Chapter3/Figures/}}

\vspace*{\fill}

\begin{Abstract}
Spermatogenesis is a recurring differentiation process that results in the production of male gametes within the testes and its in-depth characterisation is needed to understand male fertility. During spermatogenesis, spermatogonial stem cells undergo a unidirectional differentiation process to form spermatocytes, round spermatids and lastly mature sperm. This process involves a complex sequence of developmental steps and is coupled to large-scale chromatin rearrangements, therefore making it difficult to profile.
In the first instance, we thoroughly characterised spermatogenesis by   profiling transcriptomes of over 20,000 cells which were captured using droplet-based scRNA-Seq. To confidently connect transcriptional profiles to distinct developmental stages, we profiled multiple time-points during the first wave of spermatogenesis. As juvenile animals progress through spermatogenesis for the first time, development has only progressed to a certain point, thus allowing the identification of the most mature cell type. With this exact labelling, we can dissect developmental processes such as spermatogonial differentiation, meiosis and spermiogenesis on a molecular level. Furthermore, our data captured the expression dynamics of the X chromosome, which is subject to meiotic silencing in spermatocytes, followed by a partial reactivation in spermatids, revealing distinct temporal expression dynamics regarding the reactivation of the X chromosome post-meiosis. Profiling of the associated chromatin changes identified a set of genes specifically repressed by H3K9me3 in spermatocytes that later-on escape post-meiotic silencing in spermatids, demonstrating extensive chromatin remodelling on the X chromosome. After fully characterising spermatogenesis on a single-cell level, changes in transcriptional variability can be detected by estimating residual over-dispersion measures for the different germ cell populations using BASiCS. In this analysis, the differentiation trajectory defines a new confounding factor effecting the variability measures.  
\end{Abstract}

\vspace*{\fill}

\newpage

\vspace*{\fill}

\begin{Comment}
\textbf{Declaration} This project has been a collaboration including members of the Marioni and Odom lab. Christina Ernst performed all experiments presented in this chapter and performed preliminary analysis for the Cut\&{}Run data. Celia P. Martinez-Jimenez performed preliminary experiments. John Marioni and Duncan Odom supervised the study. Christina Ernst, I, John Marioni and Duncan Odom designed the study and wrote the manuscript. I performed all computational analysis and produced all figures in this chapter (except the histology images and schematics). The last section of this chapter is purely my own contribution. The preprint has been made available online at:

Christina Ernst$^\ast$, Nils Eling$^\ast$, Celia P. Martinez-Jimenez, John C. Marioni, Duncan T. Odom. Staged developmental mapping and X chromosome transcriptional dynamics during mouse spermatogenesis. \emph{bioRxiv}, 2018, ($^\ast$ equal contributions)
\end{Comment}

\vspace*{\fill}

\begin{figure}[hb]
\centering    
\includegraphics[width=\textwidth]{GraphicalAbstract.png}
\caption*{}
\end{figure}

\vspace*{\fill}


\newpage

% Include different main sections of the third chapter
\input{Chapter3/Chapt3_files/intro}
\newpage
\input{Chapter3/Chapt3_files/results}

