ze
%!TEX root = ../chapter1.tex
%******************************
%	 Introduction 
%*****************************

\section{Introduction}

Ageing is characterised by the progressive decline of physiological and cellular functions \citep{Lopez-Otin2013, Booth2016}. Nine hallmarks of ageing have been described to determine the ageing phenotype: genomic instability, telomere attrition, epigenetic alterations, loss of proteostasis, de-regulated nutrient sensing, mitochondrial dysfunction, cellular senescence, stem cell exhaustion, and altered intercellular communication \citep{Lopez-Otin2013}. Ageing can have a complex and tissue-specific impact on gene expression levels \citep{Zahn2007}, as seen by microarray expression analyses of collections of mouse CD4\plus{} and CD8\plus{} T cells \citep{Mirza2011}, rat hepatocytes \citep{Tollet-Egnell2000}, mouse and human brain \citep{Lu2004, Lee2000}, human muscle \citep{Welle2003, Zahn2006}, human kidney \citep{Rodwell2004}, human retina \citep{Yoshida2002}, and different species of Drosophila and Caenorhabditis \citep{Mccarroll2004}. For instance, ageing affects distinct functional pathways, even in closely related CD4\plus{} and CD8\plus{} T cells \citep{Mirza2011}. \\

Approaches that analyse the expression of sets of genes on a single-cell basis have more recently suggested that ageing may also alter the cell-to-cell variability of gene expression. Transcriptional noise, RNA processing aberrations, impaired DNA repair, and chromosomal instability can be caused by epigenetic changes in DNA methylation, histone modifications and chromatin remodelling \citep{Lopez-Otin2013}. DNA methylation slightly decreases globally but increases in common disease-related genes over the lifespan of humans \citep{Talens2012}. In mice, around 35\% of assayed genes showed either increased or decreased DNA methylation over ageing, with substantial tissue-specificity \citep{Maegawa2010}. Similarly, ageing introduces changes in histone modifications such as the increase of \gls{H4K16ac}, \gls{H4K20me3} and \gls{H3K4me3}, as well as the decrease of \gls{H3K9me3} or \gls{H3K27me3} \citep{Han2012, Fraga2007}. One well studied system that controls cellular function is the \emph{\gls{Sir}2} histone deacetlyase, which is encoded by seven homologs in mammals \citep{Houtkooper2016}. The chromatin-associated protein SIRT6 in mice has been shown to protect genomic stability by promoting resistance to DNA damage. Loss of this protein induces ageing-realted phenotypes within 4 weeks of murine lifespan \cite{Mostoslavsky2006}. Similar effects can be seen for SIRT1 \cite{Oberdoerffer2008}.\\

While most studies focused on identifying age-associated gene expression profiles \citep{DeMagalhaes2009}, the role of transcriptional noise during ageing has only been sporadically assessed. Analysis of fifteen genes in terminally differentiated cardiomyocytes suggested that ageing can lead to increased cell-to-cell transcriptional variability \citep{Bahar2006}. In contrast, single-cell analysis of the transcription of six genes in four different haematopoietic stem cell types showed few cell-to-cell changes between old and young animals, leading to the suggestion that transcriptional instability may not be a universal attribute of ageing \citep{Warren2007}. Whether cell-to-cell gene expression variability increases during ageing on a genome-wide basis, particularly for dynamic activation programs, remains largely unexplored.\\

Single-cell RNA sequencing presents a powerful technology to quantify  transcriptional variability in thousands of genes across thousands of cells simultaneously. For example, Kowalczyk \textit{et al.} performed a high-resolution scRNA-seq analysis of haematopoietic stem cells in young and old mice. Here, cell cycle is the primary driver for cell-to-cell variability in gene expression, and ageing decreases the entry of long-term haematopoietic stem cells into G1 phase in a cell-type-specific manner \citep{Kowalczyk2015}.\\ 

To evaluate the impact of ageing on gene expression levels and cell-to-cell transcriptional variability, we selected CD4\plus{} T cells as model system. As explained in \textbf{Box 1} (page \pageref{box1}), transcriptional noise is defined as cell-to-cell variability in expression within a homogeneous population of cells. Naive CD4\plus{} T cells are readily isolated as single, phenotypically homogeneous cells when purified from young and aged spleens and can be easily stimulated into a physiologically-relevant activated transcriptional state \emph{in vitro}. Furthermore, they are maintained in a quiescent state, but have the ability to respond to antigen stimulation with proliferation and effector differentiation, which is essential for life-long maintenance of adaptive immune function against infection and cancer \citep{Swain2012, Kim2014a}. With this, they sit at the root of adaptive immunity and disruption of their transcriptional programme can lead to severe phenotypes during ageing. \\

Previously, comparing gene expression levels in matched tissues from different mammalian species was used as a tool for revealing conserved cell-type-specific regulatory programs \citep{Sudmant2015, Finseth2014, Brawand2011, Flajnik2009}. For instance, a conserved set of response genes has been identified by comparison of bulk gene expression between human and mouse CD4\plus{} T cells after immune activation \citep{Shay2013}. So far, it is not known whether conservation of gene expression levels is also reflected in cell-to-cell variability.\\

Here, we dissected the activation dynamics of naive CD4\plus{} T cells at the single cell level during ageing in two sub-species of mice. With this, we assayed transcriptional dynamics during immune response and how ageing possibly perturbs this system. By comparing our findings across divergent strains of mice, we assessed the evolutionary conservation of the immune response and ageing phenotype. Furthermore, we isolated pure naive and effector memory CD4\plus{} T cells to profile age-related changes in different CD4\plus{} T cell subsets.
